src/main/java/com/artipie/YamlStorageSettings.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import com.amihaiemil.eoyaml.StrictYamlMapping;
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.asto.Storage;
+import com.artipie.asto.fs.FileStorage;
+import com.artipie.asto.s3.S3Storage;
+import io.vertx.reactivex.core.Vertx;
+import java.net.URI;
+import java.nio.file.Path;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+
+/**
+ * Storage settings built from YAML.
+ *
+ * @since 0.2
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+final class YamlStorageSettings {
```

Comment:
@olegmoz this class is actually not settings, it represent storage. Can we rename it to `YamlStorage` or `YamlStorageFrom`

Reply:
@Vatavuk I'd prefer existing name, `YamlStorage` and `YamlStorageFrom` sound like `Storage` implementations

Reply:
@olegmoz ideally it should be `Storage` implementation. This class basically takes settings and builds `Storage` object. This class should represent `Storage` built from `settings` not other way around.

Reply:
@Vatavuk I am not sure that it would be better design, so I'd prefer to keep the classes as they are

Reply:
@g4s8 I don't agree with @olegmoz but if you do we can merge, it's not a big issue.

Reply:
@g4s8 ping